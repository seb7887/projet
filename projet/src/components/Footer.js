import React from "react";
import styled from "styled-components";

const Div = styled.div`
  text-align: center;

  @media (max-width: 1200px) {
    display: none;
  }
`;

const Link = styled.a`
  text-decoration: none;
  font-weight: bold;
  color: #f1f2f1;
  cursor: pointer;
`;

class Footer extends React.Component {
  shouldComponentUpdate(nextProps, nextState) {
    return false;
  }

  render() {
    return (
      <Div>
        <p>
          Built by Sebastian Matias Segura. Copyright &copy;2018 by{" "}
          <Link href="https://github.com/seb7887">seb7887</Link>
        </p>
      </Div>
    );
  }
}

export default Footer;
