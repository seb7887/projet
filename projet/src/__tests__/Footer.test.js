import React from "react";
import { shallow } from "enzyme";
import Footer from "../components/Footer";

describe("Footer component", () => {
  it("expect to render Footer component", () => {
    expect(shallow(<Footer />)).toMatchSnapshot();
  });

  it("expect not to be updated", () => {
    const comp = shallow(<Footer />);
    const shouldUpdate = comp.instance().shouldComponentUpdate();
    expect(shouldUpdate).toBe(false);
  });
});
