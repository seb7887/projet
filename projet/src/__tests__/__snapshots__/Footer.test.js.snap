// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component expect to render Footer component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Built by Sebastian Matias Segura. Copyright ©2018 by 
        <a
                href="https://github.com/seb7887"
                style={
                        Object {
                                "color": "#fdfcfb",
                                "cursor": "pointer",
                                "fontWeight": "bold",
                                "textDecorationLine": "none",
                              }
                }
        >
                seb7887
        </a>
</p>,
      "style": Object {
        "borderTop": "1px solid #00000",
        "paddingTop": "2rem",
        "textAlign": "center",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Built by Sebastian Matias Segura. Copyright ©2018 by ",
          <a
            href="https://github.com/seb7887"
            style={
                        Object {
                                    "color": "#fdfcfb",
                                    "cursor": "pointer",
                                    "fontWeight": "bold",
                                    "textDecorationLine": "none",
                                  }
            }
>
            seb7887
</a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Built by Sebastian Matias Segura. Copyright ©2018 by ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "seb7887",
            "href": "https://github.com/seb7887",
            "style": Object {
              "color": "#fdfcfb",
              "cursor": "pointer",
              "fontWeight": "bold",
              "textDecorationLine": "none",
            },
          },
          "ref": null,
          "rendered": "seb7887",
          "type": "a",
        },
      ],
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Built by Sebastian Matias Segura. Copyright ©2018 by 
          <a
                    href="https://github.com/seb7887"
                    style={
                              Object {
                                        "color": "#fdfcfb",
                                        "cursor": "pointer",
                                        "fontWeight": "bold",
                                        "textDecorationLine": "none",
                                      }
                    }
          >
                    seb7887
          </a>
</p>,
        "style": Object {
          "borderTop": "1px solid #00000",
          "paddingTop": "2rem",
          "textAlign": "center",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Built by Sebastian Matias Segura. Copyright ©2018 by ",
            <a
              href="https://github.com/seb7887"
              style={
                            Object {
                                          "color": "#fdfcfb",
                                          "cursor": "pointer",
                                          "fontWeight": "bold",
                                          "textDecorationLine": "none",
                                        }
              }
>
              seb7887
</a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Built by Sebastian Matias Segura. Copyright ©2018 by ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "seb7887",
              "href": "https://github.com/seb7887",
              "style": Object {
                "color": "#fdfcfb",
                "cursor": "pointer",
                "fontWeight": "bold",
                "textDecorationLine": "none",
              },
            },
            "ref": null,
            "rendered": "seb7887",
            "type": "a",
          },
        ],
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
